### ISSUE TITLE
Failure installing Microsoft.EntityFrameworkCore.Relational.Design.Specification.Tests package

### ISSUE THREAD

**Issue by ErikEJ** [aspnet/EntityFramework#5034]
I am trying to install build 20507 from the aspnetrelease feed.
I am using VS 2015 Update 2 with NuGet 3.4.2

When trying to install the above package in a .NET 4.6 .csproj based Class library project, I get the following error (I am able to install the EFCore.Relational.Design package to the same project without errors):

**Could not install package 'runtime.native.System 4.0.0-rc2-23931'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.6', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.**

This prevents me from running E2E tests of the reverse Engineering component in the SQL CE provider.

**Comment by rowanmiller** [comment]
@smitpatel this may go away with your changes, can you verify

**Comment by natemcmaster** [comment]
@divega this looks similar to the issue you saw with Microsoft.AspNet.Identity.EntityFramework. Did we every figure out if there is a NuGet bug with fx packages?

**Comment by smitpatel** [comment]
This still persists with new packages.

**Comment by divega** [comment]
@natemcmaster I am not sure it looks the same, but the error you get from NuGet is misleading so hard to tell. 

In case it helps, NuGet does not support packages with references to design-time facades, and apparently that is by design. For some reason it doesn't always fails though. See https://github.com/NuGet/Home/issues/2193 and https://github.com/NuGet/Home/issues/2594 for more examples.

**Comment by divega** [comment]
@natemcmaster I reproed it myself and I actually believe that you are right. Not sure it is relevant but the last package that installs successfully is Remotion.Linq.2.0.2 and after that it tries to find this package that doesn't exist.

**Comment by divega** [comment]
In case it helps, I looked at the logs of installing this on both on .NET 4.6 and on .NET 4.5.1. 

At the point in which you start getting failures on .NET 4.6, in .NET 4.5.1 (where installing this package succeeds) the next package that gets installed is: System.Collections.4.0.11-rc2-24018. 

For Microsoft.AspNet.Identity.EntityFramework the culprit turned out to be System.Diagnostics.DiagnosticSource. 

I wonder if this is a problem in the corefx packages.

**Comment by ErikEJ** [comment]
Ah, good old System.Collections!

**Comment by divega** [comment]
@ErikEJ @smitpatel Although I had originally been able to repro this I have seen Microsoft.EntityFrameworkCore.Relational.Design.Specification.Tests from both aspnetcidev and aspnetrelease install correctly in the last couple of days. Not sure if it had anything to do with an improvement in our packages, or with cleaning up the package cache in my machine. Have you observed the same?

**Comment by ErikEJ** [comment]
@divega I will attempt again next time I refresh rc2 packages. How does one: clean up the package cache?

**Comment by smitpatel** [comment]
I am still getting error, even after clearing out all packages & caches.

**Comment by divega** [comment]
There is a cache under `%userprofile%\.nuget`. You can delete that directory directly o if you have the nuget command line tool you can try `nuget locals all -clear`.

**Comment by divega** [comment]
@smitpatel with NuGet 3.4.2?

**Comment by smitpatel** [comment]
Upgraded NuGet to Version: 3.4.3.855 and tried again but still hitting error.

**Comment by ErikEJ** [comment]
@smitpatel Bummer!

**Comment by divega** [comment]
@ericstj (directing this to the other Eric on purpose :smile:) while investigating this issue we realized that the Roslyn package doesn't install in NuGet on 4.6 projects. That applies even to the latest packages from the Roslyn-nightly build:

```
Install-Package -Source https://www.myget.org/F/roslyn-nightly/api/v3/index.json -pre  Microsoft.CodeAnalysis.Common
```
Output:
```
Attempting to gather dependency information for package 'Microsoft.CodeAnalysis.Common.1.3.0-beta1-20160426-03' with respect to project 'ConsoleApplication15', targeting '.NETFramework,Version=v4.6'
Attempting to resolve dependencies for package 'Microsoft.CodeAnalysis.Common.1.3.0-beta1-20160426-03' with DependencyBehavior 'Lowest'
Resolving actions to install package 'Microsoft.CodeAnalysis.Common.1.3.0-beta1-20160426-03'
Resolved actions to install package 'Microsoft.CodeAnalysis.Common.1.3.0-beta1-20160426-03'
Adding package 'Microsoft.CodeAnalysis.Analyzers.1.1.0' to folder 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages'
Added package 'Microsoft.CodeAnalysis.Analyzers.1.1.0' to folder 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages'
Added package 'Microsoft.CodeAnalysis.Analyzers.1.1.0' to 'packages.config'
Executing script file 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\tools\install.ps1'
Successfully installed 'Microsoft.CodeAnalysis.Analyzers 1.1.0' to ConsoleApplication15
Install failed. Rolling back...
Package 'runtime.native.System.4.0.0-rc2-24027' does not exist in project 'ConsoleApplication15'
Executing script file 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\tools\uninstall.ps1'
Removed package 'Microsoft.CodeAnalysis.Analyzers.1.1.0' from 'packages.config'
Package 'runtime.native.System.4.0.0-rc2-24027' does not exist in folder 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages'
Removing package 'Microsoft.CodeAnalysis.Analyzers.1.1.0' from folder 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages'
Removed package 'Microsoft.CodeAnalysis.Analyzers.1.1.0' from folder 'C:\Users\prkrishn\documents\visual studio 14\Projects\ConsoleApplication15\packages'
Install-Package : Could not install package 'runtime.native.System 4.0.0-rc2-24027'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.6', but the package does 
not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.
``` 

Seems pretty bad. Any thoughts?

cc @pranavkm

**Comment by pranavkm** [comment]
cc @emgarten

**Comment by ericstj** [comment]
> Install-Package : Could not install package 'runtime.native.System 4.0.0-rc2-24027'. You are trying to install this package into a project that targets '.NETFramework,Version=v4.6', but the package does 
not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.

That's been fixed in the latest RC3 builds.  I did two things to fix it: 1) reduced the places where we have dependencies on the UNIX shims (runtime.native.*) by harvesting these dependencies by crawling the IL for p-invokes, similar to how we harvest the assembly refs; 2) made all the shim packages install-able via packages config by adding placeholders.

Even without those fixes upstack packages can avoid this by reducing their dependency groups for frameworks with inbox contracts.  EG: net45 group that is empty, netstandard1.1 group that references System.Runtime et.al.

**Comment by pranavkm** [comment]
> EG: net45 group that is empty, netstandard1.1 group that references System.Runtime et.al.

ELI5 please

**Comment by ericstj** [comment]
@pranavkm have a look at the dependencies for System.Collections.Immutable.

```
      <group targetFramework="MonoAndroid1.0" />
      <group targetFramework="MonoTouch1.0" />
      <group targetFramework=".NETFramework4.5" />
      <group targetFramework=".NETStandard1.0">
        <dependency id="System.Collections" version="4.0.11-rc3-24026-00" />
        <dependency id="System.Diagnostics.Debug" version="4.0.11-rc3-24026-00" />
        <dependency id="System.Globalization" version="4.0.11-rc3-24026-00" />
        <dependency id="System.Linq" version="4.1.0-rc3-24026-00" />
        <dependency id="System.Resources.ResourceManager" version="4.0.1-rc3-24026-00" />
        <dependency id="System.Runtime" version="4.1.0-rc3-24026-00" />
        <dependency id="System.Runtime.Extensions" version="4.1.0-rc3-24026-00" />
        <dependency id="System.Threading" version="4.0.11-rc3-24026-00" />
      </group>
      <group targetFramework="Windows8.0" />
      <group targetFramework="WindowsPhone8.0" />
      <group targetFramework="WindowsPhoneApp8.1" />
      <group targetFramework="Xamarin.iOS1.0" />
      <group targetFramework="Xamarin.Mac2.0" />
      <group targetFramework="Xamarin.TVOS1.0" />
      <group targetFramework="Xamarin.WatchOS1.0" />
```
NETStandard group has package dependencies.  Every other group has empty dependency groups since all the packages have inbox implementations.

This isn't a requirement, as I've mentioned we've fixed the CoreFx packages to work in more places (like desktop projects using packages.config), but it does reduce noise in packages.config.

**Comment by ErikEJ** [comment]
Exact same issue with build rc2-20801

**Comment by divega** [comment]
Long story short, we confirmed this to be an issue in the Roslyn packages and its dependencies but it won't be fixed for RC2. It is already fixed in post-RC2 bits.

@ErikEJ does this block you or can you target 4.5.1? Otherwise @ericstj mentioned you can use project.json as a workaround.

**Comment by divega** [comment]
We should verify that this is fixed for 1.0.0.

**Comment by ErikEJ** [comment]
@divega At some point in history, I was forced to target .NET 4.6 from the test project, due to some dependency - after having reverted to .NET 4.5.1, I am now UNBLOCKED and all tests pass! Thanks...

**Comment by smitpatel** [comment]
still issue 😢

**Comment by divega** [comment]
It seems the Roslyn packages we are getting are still dependent on older RC2 packages of .NET Core. The fix is in RC3 of .NET Core, but not sure when the dependencies will be upgraded. @pranavkm may know.

**Comment by pranavkm** [comment]
They're working on updating to RC3. We might have to wait until Tuesday to find out if this is fixed.

**Comment by divega** [comment]
Still broken but Microsoft.CodeAnalysis.Common from the roslyn-stabilization-nightly feed now works, so hopefully we'll get the updated dependencies soon.

cc @pranavkm

**Comment by pranavkm** [comment]
Still waiting on the CLI to update. Not super sure why they haven't updated as yet - https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.Compiler.Common/project.json#L7

**Comment by smitpatel** [comment]
.NET 4.6 class library
with aspnetcidev as package feed
`Install-Package Microsoft.entityframeworkcore.relational.design.specification.tests -pre` installs the package correctly.
The package version is `1.0.0-rc3-21243`

**Comment by smitpatel** [comment]
@divega - close this now?

**Comment by divega** [comment]
Thanks @smitpatel!

**Comment by ErikEJ** [comment]
Thanks, guys!
